pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -98,25 +104,39 @@ func (fr *FileReader) walkDisk() error {
 
 	fr.handleMissingDashboardFiles(provisionedDashboardRefs, filesFoundOnDisk)
 
-	sanityChecker := newProvisioningSanityChecker(fr.Cfg.Name)
-
+	usageTracker := newUsageTracker()
 	if fr.FoldersFromFilesStructure {
-		err = fr.storeDashboardsInFoldersFromFileStructure(filesFoundOnDisk, provisionedDashboardRefs, resolvedPath, &sanityChecker)
+		err = fr.storeDashboardsInFoldersFromFileStructure(filesFoundOnDisk, provisionedDashboardRefs, resolvedPath, usageTracker)
 	} else {
-		err = fr.storeDashboardsInFolder(filesFoundOnDisk, provisionedDashboardRefs, &sanityChecker)
+		err = fr.storeDashboardsInFolder(filesFoundOnDisk, provisionedDashboardRefs, usageTracker)
 	}
 	if err != nil {
 		return err
 	}
 
-	sanityChecker.logWarnings(fr.log)
+	fr.mux.Lock()
+	defer fr.mux.Unlock()
 
+	fr.usageTracker = usageTracker
 	return nil
 }
 
+func (fr *FileReader) changeWritePermissions(restrict bool) {
+	fr.mux.Lock()
+	defer fr.mux.Unlock()
+
+	fr.dbWriteAccessRestricted = restrict
+}
+
+func (fr *FileReader) isDatabaseAccessRestricted() bool {
+	fr.mux.RLock()
+	defer fr.mux.RUnlock()
+
+	return fr.dbWriteAccessRestricted
+}
+
 // storeDashboardsInFolder saves dashboards from the filesystem on disk to the folder from config
-func (fr *FileReader) storeDashboardsInFolder(filesFoundOnDisk map[string]os.FileInfo,
-	dashboardRefs map[string]*models.DashboardProvisioning, sanityChecker *provisioningSanityChecker) error {
+func (fr *FileReader) storeDashboardsInFolder(filesFoundOnDisk map[string]os.FileInfo, dashboardRefs map[string]*models.DashboardProvisioning, usageTracker *usageTracker) error {
```

Comment:
It's better if you don't make the line too long, since it's then difficult to diff.
```suggestion
func (fr *FileReader) storeDashboardsInFolder(filesFoundOnDisk map[string]os.FileInfo, 
	dashboardRefs map[string]*models.DashboardProvisioning, usageTracker *usageTracker) error {
```